hip-service/Link/Patient/LinkPatient.cs

```diff
@@ -0,0 +1,116 @@
+        private readonly IPatientVerification _patientVerification;
+
+        public LinkPatient(ILinkPatientRepository linkPatientRepository, PatientRepository patientRepository,
+            IPatientVerification patientVerification)
+        {
+            this.linkPatientRepository = linkPatientRepository;
+            _patientRepository = patientRepository;
+            _patientVerification = patientVerification;
+        }
+
+        public async Task<Tuple<PatientLinkReferenceResponse, Error>> LinkPatients(PatientLinkReferenceRequest request)
+        {
+
+            var patient = request.Patient;
+            
+            if (_patientRepository.GetPatientInfoWithReferenceNumber(patient.ReferenceNumber) == null)
+            {
+                return new Tuple<PatientLinkReferenceResponse, Error>
+                    (null, new Error(ErrorCode.NoPatientFound, "No patient found"));
+            }
+            var programs = new List<hip_service.Discovery.Patient.models.CareContext>();
+            foreach (var careContext in patient.CareContexts)
+            {
+                programs.Add(_patientRepository.GetProgramInfo(patient.ReferenceNumber,
+                    careContext.ReferenceNumber));
+            }
+
+            if (programs.Count != patient.CareContexts.Count())
+            {
+                return new Tuple<PatientLinkReferenceResponse, Error>
+                    (null, new Error(ErrorCode.CareContextNotFound, "Care context not found for given patient"));
+            }           
+            
+            var linkRefNumber = Guid.NewGuid().ToString();
+            
+            // method call for generating the OTP
+            var session = new Session(linkRefNumber, new Communication(IdentifierType.MOBILE,""));
+
+            if (await _patientVerification.GenerateVerificationToken(session) != null)
+            {
+                return new Tuple<PatientLinkReferenceResponse, Error>
+                    (null, new Error(ErrorCode.OtpInValid, "Unable to create token"));
+            }
+            
+            var (_, exception) = await linkPatientRepository.SaveLinkPatientDetails(linkRefNumber, patient.ConsentManagerId,
+                patient.ConsentManagerUserId, patient.ReferenceNumber, patient.CareContexts
+                    .Select(context => context.ReferenceNumber).ToArray());
+
+            if (exception != null)
+            {
+                return new Tuple<PatientLinkReferenceResponse, Error>
+                    (null, new Error(ErrorCode.CareContextNotFound, "Unable to store data to Database"));
+            }
+            
+            var expiry = new DateTime (DateTime.Today.Year, DateTime.Today.Month, DateTime.Today.Day, 
+                    DateTime.Now.Hour, DateTime.Now.Minute+1, DateTime.Now.Second).ToUniversalTime().
+                ToString("yyyy-MM-ddTHH:mm:ssZ"); 
+
+            var meta = new LinkReferenceMeta(nameof(IdentifierType.MOBILE).ToUpper(), 
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Define the locale to be used in this string operation.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4769144)